core/src/main/java/org/jruby/RubyException.java

```diff
@@ -335,9 +335,9 @@ public IRubyObject cause(ThreadContext context) {
      * @return the converted result
      */
     @Override
-    public Object toJava(Class target) {
-        if (target.isAssignableFrom(RaiseException.class)) {
-            return target.cast(throwable);
+    public <T> T toJava(Class<T> target) {
+        if (target != Object.class && target.isAssignableFrom(RaiseException.class)) {
```

Comment:
Perhaps we should widen this to Thowable? Right now that, Exception, and RuntimeException all will return self, which is not any of those.

Reply:
not sure if I got that right, but passing `Throwable.class` returns the `RaiseException`
... see test which is part of the diff: https://github.com/jruby/jruby/pull/5488/commits/23934bfabc5ef4cada58a4901c38da4db01dd80c#diff-f9f68150a69d43ace7536ed08c60f821R68

my brain has always a hard-time reasoning `target.isAssignable(...)` but it does the job here.
`RubyException` isn't a `Throwable` so this is the best we get from `toJava(Throwable.class)`

Reply:
Ok, so basically anything you could assign a RaiseException value will return the throw able, *except* for Object.class in your patch now returns self. Yeah, isAssignableFrom is badly named...I always have to reread the docs.

Reply:
> so basically anything you could assign a RaiseException value will return the throw able, except for Object.class in your patch now returns self.

yep that's the regression, Nokogiri stores RubyException instances in a `RubyArray` and previously a `List#get` (`element.toJava(Object.class)`) on such array returned the instance itself.